great_expectations/datasource/sqlalchemy_datasource.py

```diff
@@ -161,20 +161,26 @@ def get_batch(self, batch_kwargs, batch_parameters=None):
             batch_reference = SqlAlchemyBatchReference(engine=self.engine, query=query, table_name=query_support_table_name,
                                                        schema=batch_kwargs.get("schema"))
         elif "table" in batch_kwargs:
+            table = batch_kwargs['table']
             limit = batch_kwargs.get('limit')
             offset = batch_kwargs.get('offset')
             if limit is not None or offset is not None:
                 logger.info("Generating query from table batch_kwargs based on limit and offset")
+                # In BigQuery the table name is already qualified with its schema name
+                if self.engine.dialect.name.lower() == "bigquery":
+                    schema = None
+                else:
+                    schema = batch_kwargs.get("schema")
```

Comment:
How can we assert or check this given that the user can provide table in any batch kwargs, not just ones that come from the init flow?

Reply:
There are 2 cases:
1. The SQLAlchemy URL has no dataset in it ("bigquery://project_name"). This is equivalent to not setting a default dataset. In this case the user *must* qualify all their table names in queries ("dataset.table") - our logic does the right thing, because if the table name provided in batch kwargs is not fully qualifies, the query *should* fail.
2. The SQLAlchemy URL has a a dataset in it ("bigquery://project_name/dataset_name"). This dataset becomes the default for all tables in this session. If the user provides a fully qualified table name, everything works. If the user provides a short table name (without schema), it works too.

Reply:
👍 Thanks for clarifying!